src/web/demo/src/main/java/org/geoserver/web/demo/DemoRequestsPage.java

```diff
@@ -63,7 +93,30 @@ public class DemoRequestsPage extends GeoServerBasePage {
 
     private PasswordTextField password;
 
-    public DemoRequestsPage() {
+    private CheckBox prettyXML;
+    private CheckBox openNewPage;
+
+    public DemoRequestsPage(PageParameters parameters) {
+        super(parameters);
+        setup();
+        if (parameters != null) {
+            if (parameters.get("xml") != null) {
+                ((DemoRequest) this.getDefaultModel().getObject())
+                        .setRequestBody(parameters.get("xml").toString());
+            }
+            if (parameters.get("url") != null) {
+                ((DemoRequest) this.getDefaultModel().getObject())
+                        .setRequestUrl(parameters.get("url").toString());
+            }
+        }
+    }
+
+    @Override
+    public void renderHead(IHeaderResponse response) {
```

Comment:
Re-opening https://osgeo-org.atlassian.net/browse/GEOS-11390 due to missing call to super.render(response) here, required to ensure header contributions are included.